test/models/user_test.rb

```diff
@@ -990,4 +990,37 @@ def editing_self_helper
       end
     end
   end
+
+  describe '#visible_environments' do
+    test 'should check for environments permission with orgs/locs disabled' do
+      Taxonomy.expects(:locations_enabled).returns(false)
+      Taxonomy.expects(:organizations_enabled).returns(false)
+      setup_user 'view', 'environments', 'name != production'
+      all_but_production = Environment.unscoped.select { |e| e.name != 'production' }
+      assert_equal all_but_production.map(&:name), User.current.visible_environments
+    end
+
+    test 'should show the list of environments visible when only orgs are enabled' do
+      Taxonomy.expects(:locations_enabled).returns(false).at_least_once
+      setup_user 'view', 'environments', 'name = test_env_org1'
+      FactoryGirl.create(:environment, :name => 'test_env_org1',
+                         :organizations => [User.current.organizations.first],
+                         :locations => User.current.locations.all)
```

Comment:
Yes, but if it's nil then `Environment.unscoped.authorized(:view_environments)` will fail to show them to this user. The point here was to test the `if` condition in the user.rb `visible_environments` change. 

I'll try to flip `SETTINGS[:locations_enabled]` see if I can make it truly simulate what we want to test here.